File: rust/src/serde_config/structs/plugin_info.rs
use serde::{Deserialize, Serialize};
use super::common::*;

#[derive(Clone)]
#[derive(Serialize, Deserialize)]
pub enum PluginSource{
  Internal,
  External,
}

#[derive(Clone)]
#[derive(Serialize, Deserialize)]
pub struct PluginInfo{
  pub id: String,
  pub name: Text,
  pub description: Text,
  pub icon: Option<String>,
  pub version: u32,
  pub author: Option<String>,
  pub web: Option<String>,
  pub download_url: Option<String>,
  pub plugin_list: Option<String>,
  pub app_version: u32,
  pub platforms: Vec<Platform>,
}

File: lib/plugin/infinity_nikki_support/plugin_info.dart
import "package:nikki_albums/src/rust/serde_config/structs/common.dart";
import "package:nikki_albums/src/rust/serde_config/structs/plugin_info.dart";

const PluginInfo pluginConfig = PluginInfo(
  uuid: "05d5e067-ac1a-44c2-9f87-5ccaae613954",
  name: Text.translate(TranslateText(key: "infinity_nikki_support_name")),
  description: Text.translate(TranslateText(key: "infinity_nikki_support_description")),
  icon: "infinity_nikki.webp",
  version: 1,
  author: "Nikki Albums",
  web: r"nikki.ranaxro.com",
  downloadUrl: "",
  pluginList: null,
  appVersion: 0,
  platforms: Platform.values,
);